tracing-appender/src/rolling.rs

```diff
@@ -37,6 +37,19 @@ impl io::Write for RollingFileAppender {
     }
 }
 
+/// A function which creates a hourly rolling file appender. This appender implements only the
+/// `Write` trait and can be used with `NonBlocking` to provide a `MakeWriter` to be used with a
+/// tracing subscriber.
+/// # Examples
+/// ``` rust,ignore
+/// let appender = rolling::hourly("/some/path", "rolling.log");
+/// let (non_blocking_appender, _guard) = non_blocking(appender);
+///
+/// let subscriber = tracing_subscriber::fmt().with_writer(non_blocking_appender);
+///
+/// tracing::subscriber::with_default(subscriber.finish(), || {
+///     tracing::event!(tracing::Level::INFO, "Hello");
+/// });
```

Comment:
similarly, we definitely don't want to _run_ this when testing the examples, since it will create and write to the given file path...but we can still let RustDoc make sure the example code compiles:
```suggestion
/// ``` rust
/// # fn doc() {
/// let appender = rolling::hourly("/some/path", "rolling.log");
/// let (non_blocking_appender, _guard) = non_blocking(appender);
///
/// let subscriber = tracing_subscriber::fmt().with_writer(non_blocking_appender);
///
/// tracing::subscriber::with_default(subscriber.finish(), || {
///     tracing::event!(tracing::Level::INFO, "Hello");
/// });
/// # }
```
we may also want to stick a (non-hidden) `fn main()` around this in the docs, since I think it's important for users to understand that the `non_blocking` guard needs to be held for as long as the appender is in use:
```suggestion
/// ``` rust
/// fn main() {
/// # fn doc() {
/// let appender = rolling::hourly("/some/path", "rolling.log");
/// let (non_blocking_appender, _guard) = non_blocking(appender);
///
/// let subscriber = tracing_subscriber::fmt().with_writer(non_blocking_appender);
///
/// tracing::subscriber::with_default(subscriber.finish(), || {
///     tracing::event!(tracing::Level::INFO, "Hello");
/// });
/// # }
/// }
```